src/atmega2560p/com/usart.rs

```diff
@@ -0,0 +1,166 @@
+//     RustDuino : A generic HAL implementation for Arduino Boards in Rust
+//     Copyright (C) 2021  Devansh Kumar Jha, Indian Institute of Technology Kanpur
+//
+//     This program is free software: you can redistribute it and/or modify
+//     it under the terms of the GNU Affero General Public License as published
+//     by the Free Software Foundation, either version 3 of the License, or
+//     (at your option) any later version.
+//
+//     This program is distributed in the hope that it will be useful,
+//     but WITHOUT ANY WARRANTY; without even the implied warranty of
+//     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//     GNU Affero General Public License for more details.
+//
+//     You should have received a copy of the GNU Affero General Public License
+//     along with this program.  If not, see <https://www.gnu.org/licenses/>
+
+//! Functions provided to the user for ATMEGA2560P USART implementation.
+//! This file contains the println() functions in various versions which the user will
+//! use to transmit data using USART on ATMEGA2560P.
+//! This file combines all the functions in other USART source code to make useful functions.
+//! See the section 22 of ATMEGA2560P datasheet.
+//! https://ww1.microchip.com/downloads/en/devicedoc/atmel-2549-8-bit-avr-microcontroller-atmega640-1280-1281-2560-2561_datasheet.pdf
+
+/// Crates which would be used in the implementation.
+use crate::atmega2560p::com::serial::Serial;
+use crate::atmega2560p::com::usart_initialize::Usart;
+use crate::atmega2560p::com::usart_initialize::{
+    UsartDataSize, UsartModes, UsartNum, UsartParity, UsartPolarity, UsartStop,
+};
+
+/// Standard datatypes to be used
+use core::{f64, u32};
+
+/// Default setting parameters for various modes of USART in case user want's to skip them.
+/// Baud Rate.
+const BAUD: i64 = 2400;
+/// Frame Settings.
+const SIZE: UsartDataSize = UsartDataSize::Eight;
+const PARITY: UsartParity = UsartParity::No;
+const STOP: UsartStop = UsartStop::One;
+/// USART mode.
+const MODE: UsartModes = UsartModes::Normasync;
+/// Default USART number to be used.
+const NUM: UsartNum = UsartNum::Usart0;
+/// Default clock polarity mode.
+const _POLARITY: UsartPolarity = UsartPolarity::Outputrise;
+
+impl Serial {
+    /// Gives a new serial port object which can be used to control all the
+    /// USART at one place.
+    pub fn serial_new() -> Serial {
+        Serial::new()
+    }
+}
+
+impl Usart {
+    /// This function can be use to initialize with default settings.
+    /// Like Mode:Normal asynchronuous,stopbit:one,data bit:8,parity type:no
+    pub fn begin(&mut self) {
+        self.transmit_enable();
+        self.recieve_enable();
+        self.initialize(MODE, BAUD, STOP, SIZE, PARITY);
+    }
+
+    /// This function can be use to initialize with baud rate and remaining settings will be set to default
+    /// Like Mode:Normal asynchronuous,stopbit:one,data bit:8,parity type:no
```

Comment:
Please use proper English, I cannot understand the comment, maybe provide a usage comment?

Reply:
fixed